solutions/ceoi-19-AmusementPark.mdx

```diff
@@ -0,0 +1,172 @@
+---
+id: ceoi-19-AmusementPark
+source: CEOI 2019
+title: 2019 - Amusement Park
+author: Shijie Ren
+---
+
+[CF Editorial](https://codeforces.com/blog/entry/68748)
+
+<Spoiler title="Hint">
+
+We can observe that for each legal proposal, reversing the direction of its
+edges creates another legal proposal.
+
+The costs of these two proposals will be $m$ because each edge was chosen once.
+
+So the sum of costs will be $m\cdot\frac{\texttt{DAGCount}}2$,
+
+where $\texttt{DAGCount}$ represents the number of directed acyclic graphs (DAG)
+generated on the original graph.
+
+</Spoiler>
+
+<Spoiler title="Explanation">
+
+Let $\texttt{dp}[S]$ be the number of ways of construct a directed acyclic
+graphs (DAG) with vertex set $S$ , and $\texttt{indepset}[S]$ represents if $S$
+is an independent set.
+
+Since we're constructing a DAG, we can transfer the state by adding new points.
+Adding 0-out-degree points (or 0-in-degree points is OK) will be easier.
+
+Loop all non-empty subsets of S as points to be added, points are 0-out-degree
+if and only if they don't have edges with each other in this subset. That is,
+the subset is an independent set.
+
+<Spoiler title="A Brief Explanation">

```

Comment:
That's fantastic I found that the size of the sets to be unioned can be added directly at the beginning, but in mine I complicated it cuz I union all configs where P⊆the set of vertices with out-degree 0 at the beginning causes repititions
($A_S$ is an array of numbers? So is it $\sum$ instead of $\cup$?)
and how can I deal with my explanation :eyes:

Reply:
> ($A_S$ is an array of numbers? So is it $\sum$ instead of $\cup$?)

o yeah

> and how can I deal with my explanation 👀

just use mine if you think it's better?